torch/export/exported_program.py

```diff
@@ -274,17 +274,40 @@ def tensor_constants(self):
     def constants(self):
         return self._constants
 
-    def __call__(self, *args: Any, **kwargs: Any) -> Any:
-        import torch._export.error as error
-
+    def _get_flat_args_with_check(self, args, kwargs):
+        """Flatten args, kwargs using pytree, then, check specs.
+
+        Args:
+            args: List[Any] original args passed to __call__
+            kwargs: Dict[str, Any] original kwargs passed to __call
+
+        Returns:
+            A tuple of (flat_args, received_spec)
+            flat_args is flattend args / kwargs
+            received_spec is the pytree spec produced while flattening the
+            tuple (args, kwargs)
+        """
         in_spec = self.call_spec.in_spec
         if in_spec is not None:
             kwargs = reorder_kwargs(kwargs, in_spec)
-
         flat_args_with_path, received_spec = pytree.tree_flatten_with_path(
             (args, kwargs)
         )  # type: ignore[possibly-undefined]
+        self._check_input_constraints(flat_args_with_path)
+        flat_args = tuple(x[1] for x in flat_args_with_path)
+        return flat_args, received_spec
 
+    def graph_module_flat_inputs(self, args: Any, kwargs: Any) -> Any:
```

Comment:
@qihqi since you mentioned you want to land it soon, could we make temporarily make it a private API (_graph_module_flat_inputs)? In that way you can be unblocked and we can have some time later to make a formal design discussion.

Reply:
@zhxchen17  done. Please keep me in the loop when these functions get renamed to become public.